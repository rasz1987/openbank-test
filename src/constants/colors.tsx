import { ColorsType } from '../interfaces';

export const colors: ColorsType = {
  primary: '#FF0049',
  secondary: '#002B45',
  terciary: '#FFF1E5',
  white: '#FFFFFF',
  black: '#000000'
};