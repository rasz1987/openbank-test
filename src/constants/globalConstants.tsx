import { ResponseType } from '../interfaces';

export const PRUEBA_KO: string = 'pruebaKO123';
export const RESPONSE_OK: ResponseType = { status: 200 };
export const RESPONSE_KO: ResponseType = { status: 401 };