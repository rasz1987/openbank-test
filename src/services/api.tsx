import {
	PRUEBA_KO,
	RESPONSE_OK,
	RESPONSE_KO
} from '../constants/globalConstants';
import { SubmitFormProp } from '../interfaces';

export const submitForm: SubmitFormProp = (pass, repass, optionalQuestion) =>
	new Promise((resolve, reject) =>
		setTimeout(() =>
			pass !== PRUEBA_KO
				? resolve(RESPONSE_OK)
				: reject(RESPONSE_KO)
			, 3000)
	)
