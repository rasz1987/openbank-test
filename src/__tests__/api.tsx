import {
  RESPONSE_KO,
  RESPONSE_OK,
  PRUEBA_KO
} from '../constants/globalConstants'
import { submitForm } from '../services/api';
import { colors } from '../constants/colors';

jest.setTimeout(20000);

test('should return status 200', async () => {
  await submitForm('myPass', 'myPass', 'Question')
    .then(res => {
      expect(res.status).toBe(RESPONSE_OK.status)
    });
});

test('should return status 401', async () => {
  await submitForm(PRUEBA_KO, PRUEBA_KO, 'Question').catch(error => {
    expect(error.status).toBe(RESPONSE_KO.status);
  })
});

test('should retrun colors', () => {
  expect(colors).toMatchSnapshot();
});