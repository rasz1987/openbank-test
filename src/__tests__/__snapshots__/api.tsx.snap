// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrun colors 1`] = `
Object {
  "black": "#000000",
  "primary": "#FF0049",
  "secondary": "#002B45",
  "terciary": "#FFF1E5",
  "white": "#FFFFFF",
}
`;
